diff --git i/build/lib/Makefile w/build/lib/Makefile
new file mode 100644
index 0000000..1e72e9b
--- /dev/null
+++ w/build/lib/Makefile
@@ -0,0 +1,5 @@
+../.config:
+	$(MAKE) -C ../.. config T=x86_64-native-linuxapp-gcc
+
+%.a: ../.config
+	$(MAKE) -C ../..
diff --git i/config/common_base w/config/common_base
index 7830535..ed5fc0c 100644
--- i/config/common_base
+++ w/config/common_base
@@ -410,7 +410,7 @@ CONFIG_RTE_RING_PAUSE_REP_COUNT=0
 # Compile librte_mempool
 #
 CONFIG_RTE_LIBRTE_MEMPOOL=y
-CONFIG_RTE_MEMPOOL_CACHE_MAX_SIZE=512
+CONFIG_RTE_MEMPOOL_CACHE_MAX_SIZE=4096
 CONFIG_RTE_LIBRTE_MEMPOOL_DEBUG=n
 
 #
