diff --git i/lib/log.c w/lib/log.c
index 15d5086..1e89fe6 100644
--- i/lib/log.c
+++ w/lib/log.c
@@ -62,7 +62,6 @@
 
 static FILE *m_file = NULL;             /**< file that log writes to */
 static int m_opt = 0;                   /**< log options */
-static int m_fd = -1;			/**< log file descriptor */
 /**
  * ---------------------------------------
  * Local functions
@@ -77,22 +76,20 @@ static int m_fd = -1;			/**< log file descriptor */
  */
 
 int
-log_init(int fd, int opt)
+log_init(FILE *f, int opt)
 {
         int ret = LOG_RETVAL_OK;
 
         if (m_file != NULL)
                 return LOG_RETVAL_ERROR;
 
-        m_file = fdopen(fd, "a");
+        m_file = f;
         if (m_file == NULL) {
-                fprintf(stderr, "%s: fdopen() failed: %s\n",
-                       __func__, strerror(errno));
+                fprintf(stderr, "log_init expected a non-NULL FILE *\n");
                 return LOG_RETVAL_ERROR;
         }
 
         m_opt = opt;
-		m_fd = fd;
         return ret;
 }
 
@@ -103,12 +100,11 @@ int log_fini(void)
         if (m_file == NULL)
                 return LOG_RETVAL_ERROR;
 
-        if ((m_fd != STDOUT_FILENO) && (m_fd != STDERR_FILENO))
-                close(m_fd);
+        if ((m_file != stdout) && (m_file != stderr))
+                fclose(m_file);
 
 	m_file = NULL;
         m_opt = 0;
-	m_fd = -1;
 
         return ret;
 }
diff --git i/lib/log.h w/lib/log.h
index 97572bf..94af8f4 100644
--- i/lib/log.h
+++ w/lib/log.h
@@ -72,7 +72,7 @@ extern "C" {
  * @return Operation status
  * @retval LOG_RETVAL_OK on success
  */
-int log_init(int fd, int opt);
+int log_init(FILE *f, int opt);
 
 /**
  * @brief Shuts down PQoS log module
diff --git i/lib/pqos.h w/lib/pqos.h
index 7f7ee3f..ed7bd74 100644
--- i/lib/pqos.h
+++ w/lib/pqos.h
@@ -88,7 +88,7 @@ enum pqos_cdp_config {
  * PQoS library configuration structure
  */
 struct pqos_config {
-        int fd_log;                     /**< file descriptor to be used for
+        FILE *fd_log;                     /**< file descriptor to be used for
                                            library to log messages */
         int verbose;                    /**< if true increases library
                                            verbosity level */
diff --git i/main.c w/main.c
index fa6c54d..992f95a 100644
--- i/main.c
+++ w/main.c
@@ -2229,10 +2229,9 @@ int main(int argc, char **argv)
          * Set up file descriptor for message log
          */
         if (sel_log_file == NULL) {
-                cfg.fd_log = STDOUT_FILENO;
+                cfg.fd_log = stdout;
         } else {
-                cfg.fd_log = open(sel_log_file, O_WRONLY|O_CREAT,
-                                  S_IRUSR|S_IWUSR);
+                cfg.fd_log = fopen(sel_log_file, "w");
                 if (cfg.fd_log == -1) {
                         printf("Error opening %s log file!\n", sel_log_file);
                         exit_val = EXIT_FAILURE;
